src/service/service.py

```diff
@@ -156,6 +177,18 @@ async def message_generator(
                 # So we only print non-empty content.
                 yield f"data: {json.dumps({'type': 'token', 'content': convert_message_content_to_string(content)})}\n\n"
             continue
+    
+    # Handle interruptions
+    snapshot = await agent.aget_state(kwargs["config"])
+    if snapshot.next: 
+        try:
+            ai_message = langchain_to_chat_message(snapshot.values["messages"][-1])
+            ichat_message = interrupt_from_call_tool(call_tool=ai_message.tool_calls[0])
+            ichat_message.run_id = str(run_id)
+        except Exception as e:
+            yield f"data: {json.dumps({'type': 'error', 'content': f'Error parsing interrupt message: {e}'})}\n\n"
+            
+        yield f"data: {json.dumps({'type': 'interrupt', 'content': ichat_message.model_dump()})}\n\n"
```

Comment:
This only handles the `/stream` endpoint. What happens if I call `/invoke` on an interrupting agent?